app/models/manageiq/providers/openstack/infra_manager/refresh_parser.rb

```diff
@@ -79,23 +79,39 @@ def ems_inv_to_hashes
 
       private
 
-      def all_server_resources
-        return @all_server_resources if @all_server_resources
+      def stack_resources_by_depth(stack, depth = 2)
+        return @stack_resources if @stack_server_resources
+        # TODO(lsmola) loading this from already obtained nested stack hierarchy will be more effective. This is one
+        # extra API call. But we will need to change order of loading, so we have all resources first.
+        @stack_resources = @orchestration_service.list_resources(:stack => stack, :nested_depth => depth).body['resources']
+      end
+
+      def filter_stack_resources_by_resource_type(resource_type_list)
         resources = []
         stacks.each do |stack|
           # Filtering just server resources which is important to us for getting Purpose of the node
           # (compute, controller, etc.).
-          resources += all_stack_server_resources(stack).select do |x|
-            %w(OS::TripleO::Server OS::Nova::Server).include?(x["resource_type"])
+          resources += stack_resources_by_depth(stack).select do |x|
+            resource_type_list.include?(x["resource_type"])
           end
         end
-        @all_server_resources = resources
+        resources
       end
 
-      def all_stack_server_resources(stack)
-        # TODO(lsmola) loading this from already obtained nested stack hierarchy will be more effective. This is one
-        # extra API call. But we will need to change order of loading, so we have all resources first.
-        @orchestration_service.list_resources(:stack => stack, :nested_depth => 2).body['resources']
+      def stack_resource_groups
+        return @stack_resource_groups if @stack_resource_groups
+        @stack_resource_groups = filter_stack_resources_by_resource_type(["OS::Heat::ResourceGroup"])
+      end
+
+      def stack_server_resource_types
+        return @stack_server_resource_types if @stack_server_resource_types
+        @stack_server_resource_types = ["OS::TripleO::Server", "OS::Nova::Server"]
+        @stack_server_resource_types += stack_resource_groups.map { |rg| "OS::TripleO::" + rg["resource_name"] + "Server" }
```

Comment:
Is this nesting of Server under other types documented somewhere?

Reply:
hahahahahaha no, it's not

Reply:
Hah, good job figuring it out then :)